blobstore/blobstore.py

```diff
@@ -0,0 +1,42 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+class BlobStore(object):
+    """Abstract base class for all blob stores."""
+    def __init__(self):
+        pass
+
+    def list(self, prefix=None):
+        """Returns a list of all blob entries in a container that match a given prefix."""
+        raise NotImplementedError()
+
+    def get_presigned_url(self, objname):
```

Comment:
Would this be a pre-signed read URL or pre-signed write URL? If read, I suggest renaming this to `get_presigned_read_url()` for the avoidance of doubt. If write or both, we'll need a more comprehensive model of the underlying APIs here (since different object stores use different upload models).